ethcore/trace/src/db.rs

```diff
@@ -57,6 +57,16 @@ impl Key<FlatBlockTraces> for H256 {
 	}
 }
 
+/// `DatabaseExtras` provides an interface to query extra data which is not stored in TraceDB,
+/// but necessary to work correctly.
+pub trait DatabaseExtras {
```

Comment:
If we still need the `blockchain` dependenccy we can probably just implement `DatabaseExtras` for any `BlockProvider` here easily. There is no need for a wrapper struct afaict.